store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreTest.java

```diff
@@ -65,39 +67,115 @@
+    private static final String KEY_MAP_ID1_PROPERTIES_ID = "gaffer.federatedstore.mockMapGraphId1.properties.id";
+    private static final String GRAPH_IDS = "gaffer.federatedstore.graphIds";
+    private static final String KEY_ACC_ID1_SCHEMA_FILE = "gaffer.federatedstore.mockAccGraphId1.schema.file";
+    private static final String KEY_MAP_ID1_SCHEMA_FILE = "gaffer.federatedstore.mockMapGraphId1.schema.file";
+    private static final String KEY_MAP_ID1_SCHEMA_ID = "gaffer.federatedstore.mockMapGraphId1.schema.id";
+    private static final String PATH_INVALID = "nothing.json";
+    private static final String EXCEPTION_NOT_THROWN = "exception not thrown";
+    private static final String USER_ID = "testUser";
+    private static final User TEST_USER = new User.Builder().userId(USER_ID).opAuths("one", "two").build();
+    private static final String PROPS_ID_1 = "PROPS_ID_1";
+    private static final String SCHEMA_ID_1 = "SCHEMA_ID_1";
+    private static final String CACHE_SERVICE_CLASS_STRING = "uk.gov.gchq.gaffer.cache.impl.HashMapCacheService";
+    private FederatedStore store;
     private StoreProperties federatedProperties;
 
     @Before
     public void setUp() throws Exception {
         store = new FederatedStore();
         federatedProperties = new StoreProperties();
+        federatedProperties.set(CacheProperties.CACHE_SERVICE_CLASS, CACHE_SERVICE_CLASS_STRING);
         HashMapGraphLibrary.clear();
     }
 
     @Test
+    public void shouldInitialiseWithCache() throws StoreException {
+        store.initialise(FEDERATED_STORE_ID, null, federatedProperties);
+        assertNotNull(CacheServiceLoader.getService());
+    }
+
+    @Test
+    public void shouldThrowExceptionWithoutInitialisation() {
+        // Given
+        Graph graphToAdd = new Graph.Builder()
+                .config(new GraphConfig(ACC_ID_1))
+                .storeProperties(StreamUtil.openStream(FederatedStoreTest.class, PATH_ACC_STORE_PROPERTIES))
+                .addSchema(StreamUtil.openStream(FederatedStoreTest.class, PATH_BASIC_EDGE_SCHEMA_JSON))
+                .build();
+
+        // When / Then
+        try {
+            store.addGraphs(graphToAdd);
+            fail("Exception expected");
+        } catch (final StoreException e) {
+            assertTrue(e.getMessage().contains("No cache has been set"));
+            System.out.println(e);
+        }
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenInitialisedWithNoCacheClassInProperties() throws StoreException {
+        // Given
+        StoreProperties storeProperties = new StoreProperties();
+
+        try {
+            store.initialise(FEDERATED_STORE_ID, null, storeProperties);
+            fail("Exception expected");
+        } catch (final StoreException e) {
+            assertTrue(e.getMessage().contains("No cache has been set, please check the property"));
+        }
+    }
+
+    @Test
+    public void shouldAddGraphsToCache() throws StoreException {
```

Comment:
This test is not checking the cache, it does not examine that cache exists, has been modified, or the cache contains the "graphToAdd". Please revise.

Reply:
I would want to see a test that replicates the issue explained within the ticket.